src/Elastic.Apm/DiagnosticListeners/HttpDiagnosticListenerImplBase.cs

```diff
@@ -153,12 +155,15 @@ private void ProcessStartEvent(TRequest request, Uri requestUrl)
 				}
 			}
 
-			if (!RequestHeadersContain(request, TraceContext.TraceStateHeaderName) && transaction.OutgoingDistributedTracingData != null
-				&& transaction.OutgoingDistributedTracingData.HasTraceState)
-			{
-				RequestHeadersAdd(request, TraceContext.TraceStateHeaderName,
-					TraceContext.BuildTraceState(transaction.OutgoingDistributedTracingData));
-			}
+			if (!RequestHeadersContain(request, TraceContext.TraceStateHeaderName) && span.OutgoingDistributedTracingData != null && span.OutgoingDistributedTracingData.HasTraceState)
+				RequestHeadersAdd(request, TraceContext.TraceStateHeaderName, span.OutgoingDistributedTracingData.TraceState.ToTextHeader());
+
+			//if (!RequestHeadersContain(request, TraceContext.TraceStateHeaderName) && transaction.OutgoingDistributedTracingData != null
```

Comment:
nit: I think this is just some leftover, right? Could you please remove it?

Reply:
done!